### PR TITLE
implement searching

### PR DESCRIPTION
**Pull request by hngerebara**
Things done in this PR

Design to match 
[Listings view](https://app.zeplin.io/project/5d581fcb1db0449b12b76763/screen/5d58283d0b6b443505e2853d) 

[Detail view](https://app.zeplin.io/project/5d581fcb1db0449b12b76763/screen/5d58283e91c8889afa13f5f2)

[Newsletter modal view](https://app.zeplin.io/project/5d581fcb1db0449b12b76763/screen/5d58283c8d5c26520ba7472a)

[No results view](https://app.zeplin.io/project/5d581fcb1db0449b12b76763/screen/5d58283dccabfe9c037443a4)

Only the Filter dropdown [here](https://app.zeplin.io/project/5d581fcb1db0449b12b76763/screen/5d65212c9df85c9b3ca072f8)

[Table view](https://app.zeplin.io/project/5d581fcb1db0449b12b76763/screen/5d6599b8c463d49b78307757)
Implemented search and filtering by tags and cloud

To be completed today:
Google analytics

 https://92eca3a4.ngrok.io

### CODE REVIEW

File: _layouts/post.html [hunk 1]
```diff
@@ -12,47 +12,44 @@
...
-    </div>
-    </div>
+      <div class="card contact-us-card text-center">
+        <span class="glyphicon glyphicon-exclamation-sign" aria-hidden="true"></span>
+        <div class="card-body">
+          <p class="card-title">Want to learn more? Feel free to contact us for support</p>
+          <a href="#" class="btn btn-primary">Support</a>
+          <a href="#">Dismiss</a>
+        </div>
+      </div>
```
Comment (brikis98) [inline _layouts/post.html]: Are there real changes on this page or just whitespace changes? Quite hard to tell. If it's whitespace, could you tweak your editor to avoid that, so PRs are easier to review (and diffs easier to use for debugging)?

File: pages/guides/_search.html [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+                <span class="glyphicon glyphicon-search" aria-hidden="true"></span>
+            </button>
+        </div>
+    </div><input id="search-box" type="text" class="form-control" placeholder="Search guides....">
+    <script>
+        window.guideEntries = [
+            {% for post in site.posts %} 
+            {
+                "title": "{{ post.title | xml_escape }}",
+                "image": "{{ post.image }}",
```
Comment (brikis98) [inline pages/guides/_search.html]: Seems like you should call `strip_html`, `strip_newlines`, `jsonify` on each of these?

File: assets/js/search.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        
+        if(searchQuery.length === 0)  {
+            $('#filtered-records').html('');
+            $('#listings').show();
+            return;
+        } else {
+            // Hide the initial data before search began
+            $('#listings').hide();
+
+            var regex = new RegExp(searchQuery, "i");
```
Comment (brikis98) [inline assets/js/search.js]: Using user-provided data like this, unsanitized, is a bit risky. Why do we need to use RegExp here? If you lower case the data and the query, an includes check should do it, right?

File: assets/js/search.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+            var output = '<div class="row">';
```
Comment (brikis98) [inline assets/js/search.js]: Assembling HTML like this is error prone, hard to maintain, and makes injection attacks possible. 

What about adding a CSS class to elements that don't match the search query to hide them? So the page stays largely the same, but certain items hide / show? 

If you're not sure what the search UI should look like, feel free to ping Sam!

File: _layouts/post.html (hunk 1 above)
Comment (hngerebara) [reply _layouts/post.html]: Yeah, these are just white spaces. Both changes were made by me as this branch was checked out of the [translate-deployment-guides](https://github.com/gruntwork-io/gruntwork-io.github.io/tree/translate-deployment-guides)

File: assets/js/search.js (hunk 3 above)
Comment (hngerebara) [reply assets/js/search.js]: True we do not need regex at all. This file has now been modified

File: assets/js/search.js (hunk 4 above)
Comment (hngerebara) [reply assets/js/search.js]: Writing it was also hard :) meaning it will be hard to maintain. I wanted someone to give feedback.
I made use of jquery .hide() and .show() to implement this so the page remains the same

File: pages/guides/_search.html (hunk 2 above)
Comment (hngerebara) [reply pages/guides/_search.html]: Yes, done that [here](https://github.com/gruntwork-io/gruntwork-io.github.io/pull/147/files#diff-a77f40945e9bcc4da6be5f4e258b3421R4)

File: _includes/footer.html [hunk 5]
```diff
@@ -5,7 +5,7 @@
     <label>Ready to hand off the Gruntwork?</label>
   </div>
   <div>
-    <button class="btn btn-primary-rounded">Buy Now</button>
+    <a href="/contact" class="btn btn-primary-rounded" role="button" aria-pressed="true">Buy Now</a>
```
Comment (brikis98) [inline _includes/footer.html]: Why did this change?

File: _includes/newsletter-modal.html [hunk 6]
```diff
@@ -0,0 +1,12 @@
+<div class="modal fade" id="newsletter-success" tabindex="-1" role="dialog" aria-hidden="true">
```
Comment (brikis98) [inline _includes/newsletter-modal.html]: Is this modal used? Or does subscribing to the newsletter redirect to a MailChimp page?

File: _includes/newsletter.html [hunk 7]
```diff
@@ -0,0 +1,23 @@
+<div class="guides-section-white">
+    <div class="row text-center">
+        <div class="col-md-6 col-md-offset-3 newsletter" >
+            <img src="/assets/img/guides/img-newsletter.png" alt="newsletter_image" class="img-center"><br>
+            <label>Newsletter</label>
+            <p>Subscribe to the Gruntwork Newsletter to get notified when updates and new guides are released. No spam. Ever.</p>
+            <form id="newsletter-form" class="col-xs-10 col-xs-offset-1" method="post" >
```
Comment (brikis98) [inline _includes/newsletter.html]: How does this work? Shouldn't it POST to a MailChimp URL like http://gruntwork.io/newsletter?

File: _layouts/post.html [hunk 8]
```diff
@@ -1,58 +1,66 @@
...
-              <p class="card-title">Want to learn more? Feel free to contact us for support</p>
-              <a href="#" class="btn btn-primary">Support</a>
-              <a href="#">Dismiss</a>
-            </div>
-          </div>
     </div>
+    <div class="col-md-7 guides-section-white">
+      <ol class="breadcrumb text-center">
+        {% assign crumbs = page.url | remove:'/index.html' | split: '/' %}
+        <a href="/guides" ga-event-category="guide-{{ post.title | downcase}}" ga-event-action="deployment-guides">Dev Ops Guides</a>
```
Comment (brikis98) [inline _layouts/post.html]:
```suggestion
        <a href="/guides" ga-event-category="guide-{{ post.title | downcase}}" ga-event-action="deployment-guides">Production Deployment Guides</a> 
```

File: _layouts/post.html [hunk 9]
```diff
@@ -1,58 +1,66 @@
...
+          {% if forloop.last %}
+           <a href="#"> / {{ crumb | replace:'-',' ' | remove:'.html' | capitalize }}</a>
+          {% else %}
+            <a href="{{ crumb }}">/ {{ crumb | replace:'-',' ' | remove:'.html' | capitalize }}</a>
+          {% endif %}
+        {% endfor %}      
+      </ol>
+      <div class="post-title text-center">
+        <img class="img-center" src="{{ page.image }}" alt="post logo">
+        <label>{{ page.title }}</label>
```
Comment (brikis98) [inline _layouts/post.html]: Page title should be an `h1`. Using proper HTML tags is important for accessibility and SEO.

File: _layouts/post.html [hunk 10]
```diff
@@ -1,58 +1,66 @@
...
+          {% else %}
+            {{ words | divided_by:180 }} mins read.
+          {% endif %} 
+          Updated {{ page.date | date_to_string }}
+        </p>
+        {% if page.tags %}
+          {% for tag in page.tags %}
+            <span class="badge badge-pill badge-default">
+              {% capture tag-name %}{{tag | upcase }}{% endcapture %}
+              {% if tag-name=="AWS" or tag-name=="GCP"  or tag-name=="GKE" %} {{ tag-name }} {% else %} {{tag-name | capitalize}}{% endif %}</span>
```
Comment (brikis98) [inline _layouts/post.html]: Tags should be clickable and go deployment guides list page, filtered to that tag. Can be done in separate PR. Please create Jira card.

File: _posts/2019-08-26-how-to-use-gruntwork-service-catalog.adoc [hunk 11]
```diff
@@ -1,23 +1,15 @@
...
-:note-caption: :information_source:
-:important-caption: :heavy_exclamation_mark:
-:caution-caption: :fire:
-:warning-caption: :warning:
-endif::[]
-
-toc::[]
+:page-type: guide
+:page-layout: post
+:page-title: How to use the Gruntwork Service Catalog
```
Comment (brikis98) [inline _posts/2019-08-26-how-to-use-gruntwork-service-catalog.adoc]: Why are some of these using Jekyll front matter syntax but some using AsciiDoc?

File: _posts/2019-08-26-how-to-use-gruntwork-service-catalog.adoc [hunk 12]
```diff
@@ -1,23 +1,15 @@
-[[how_to_use_gruntwork_service_catalog]]
+---
+categories: Docker
+image: /assets/img/guides/service-catalog/grunty-blocks.png
+excerpt: Learn about production-grade infrastructure, Terraform, Terragrunt, Packer, Docker, immutable infrastructure, versioning for infrastructure code, automated tests for infrastructure code, and more.
+permalink: /guides/:categories/:title
```
Comment (brikis98) [inline _posts/2019-08-26-how-to-use-gruntwork-service-catalog.adoc]: Do we need to specify a permalink for every post? Seems error prone. Could we define this in the layout or Jekyll config instead?

File: assets/css/global.less [hunk 13]
```diff
@@ -860,244 +858,613 @@ a.caret-right{
...
+.no-transform {
+  transform: none;
+  box-shadow: none;
+  outline: none !important;
+}
+
+.guides-input-box::-webkit-input-placeholder { .guides-input-placeholder; }
+.guides-input-box::-moz-placeholder          { .guides-input-placeholder; }
+.guides-input-box::-ms-placeholder           { .guides-input-placeholder;}
+.guides-input-box::placeholder               { .guides-input-placeholder;}
```
Comment (brikis98) [inline assets/css/global.less]: What's this stuff?

File: assets/js/guides.js [hunk 14]
```diff
@@ -0,0 +1,64 @@
...
+  //Dismiss cta on guides page
+  $('.dismiss-cta').on('click', (e) => {
+    e.preventDefault();
+    $('.guides-cta-card').hide();
+  })
+
+  //Validate email
+  let validateEmail = (email) => {
+    let re = /^([a-zA-Z0-9_.-])+@(([a-zA-Z0-9-])+.)+([a-zA-Z0-9]{2,4})+$/;
+    return re.test(email);
```
Comment (brikis98) [inline assets/js/guides.js]: Email addresses are more complicated than this Regex. The only hard and fast rule is it contains an `@`. Instead of this, use an `input type="email"`, and let it handle validation.

File: assets/js/guides.js [hunk 15]
```diff
@@ -0,0 +1,64 @@
...
+  }
+
+  $('#email').focus(() => {
+    $('#newsletter-form input').css("border", "none");
+  })
```
Comment (brikis98) [inline assets/js/guides.js]: Seems like this could be done in pure CSS

File: assets/js/guides.js [hunk 16]
```diff
@@ -0,0 +1,64 @@
...
+      $('#newsletter-form input').css("border", "1px solid red");
+      $("#newsletter-success").modal('hide');
+
+      return false;
+    } else {
+
+      e.preventDefault();
+      $("#newsletter-success").modal('show');
+    }
+  });
```
Comment (brikis98) [inline assets/js/guides.js]: So this doesn't actually subscribe yet?

File: assets/js/search.js [hunk 17]
```diff
@@ -0,0 +1,218 @@
+/**
+ * Javascript specially for the IaC Library page.
```
Comment (brikis98) [inline assets/js/search.js]: Out of date comment

File: assets/js/search.js [hunk 18]
```diff
@@ -0,0 +1,218 @@
...
+        type: 'guideEntries'
+      };
+    }
+  }
+
+  //Function to track Google analytics
+  // function trackGA {
+  //    // Track what users are searching for via Google Analytics events
+  //     //   ga('send', 'event', "iac-lib", "search", "query", text);
+  // }
```
Comment (brikis98) [inline assets/js/search.js]: ?

File: assets/js/search.js [hunk 19]
```diff
@@ -0,0 +1,218 @@
...
+      }
+
+      showItemsCount(matches, submoduleMatches);
+    } else {
+      if (searchEntry.type == 'libraryEntries') {
+
+        showInitialItemsCount();
+        $('.table-clickable-row').show();
+
+      } else if (searchEntry.type = 'guideEntries') {
```
Comment (brikis98) [inline assets/js/search.js]: Multiple problems here:

1. Single `=` is not a comparison. Not sure how this works at all? This and the comparison above should be `===`.
1. Hard-coded `type` value. We should have constants defined for these.
1. The `filterData` function says it can be used with other pages, but it's clearly hard-coded for just these two pages. This function should be refactored to actually be a generic search function... E.g., Pass in the data to search as an argument; pass in callbacks to handle how to render search results; etc. Otherwise, we are just shoving two implementations side by side into one function, which actually makes maintenance harder.

File: assets/js/search.js [hunk 20]
```diff
@@ -0,0 +1,218 @@
...
+    let searchValue = target.val();
+
+    $('#no-matches').hide();
+
+    if (searchValue && searchValue.length > 0) {
+
+      let lowerText = searchValue.toLowerCase();
+      // if(typeof(searchQueries) == 'string'){
+      //   searchQueries.push(lowerText);
+      // }
```
Comment (brikis98) [inline assets/js/search.js]: ?

File: _includes/newsletter.html (hunk 7 above)
Comment (hngerebara) [reply _includes/newsletter.html]: This is outdated. I hadn't pushed it

**Comment by brikis98** [comment]
One more: 

Lists (whether numbered, unnumbered, or definition) should probably not be indented, as we use a lot of lists, and it makes the text hard to follow as you scroll as it keeps jumping left and right. Might want to check with Sam on this one.

File: _includes/newsletter-modal.html (hunk 6 above)
Comment (Etiene) [reply _includes/newsletter-modal.html]: Not yet used, but it looks like it's the modal as per design here: 
https://projects.invisionapp.com/d/main/default/#/console/17881048/379066111/preview

Comment (Etiene) [reply _includes/newsletter-modal.html]: But for current simplicity, since functionality is not implemented here, and passing this PR faster, I would also recommend just redirecting to our mailchimp page when clicking on `subscribe` and work on newsletter specifics at a later stage

File: _layouts/post.html (hunk 10 above)
Comment (Etiene) [reply _layouts/post.html]: +1 making these clickeable in a separate PR. 

Concerning this specific code, why are there so many ifs? Is it because "AWS" etc is already upper case? One danger here is that we are not going to have full control of all the tags. What if someone adds a blog post with new tag that's also an acronym? Maybe we should expect them to be properly capitalized at the metadata of the posts? And then downcase as necessary (if we use it in urls for example)?

File: _layouts/post.html [hunk 21]
```diff
@@ -1,58 +1,66 @@
...
+      </div>
+      <div class="content">
+          {{ content }}
+      </div>
     </div>
+    <div class="col-sm-2-5">
+      <div class="card guides-cta-card text-center">
+        <img src="/assets/img/grunty-cta.png" alt="newsletter_image" class="img-center"><br>
+        <div class="card-body">
+          <p class="card-title">Hand off the gruntwork, we can deploy this service for you. </p>
```
Comment (Etiene) [inline _layouts/post.html]: There are many differences between this element and it's design.

1. Position of grunty (should be higher)
2. Quality of grunty image (should be higher)
3. Width of box (should be narrower)
3. Box shadow (should be more difused)
4. Font weight ((should be thicker)
5. Font color (should be gray, not black)
6. Button color (different shade of blue)


Expected:
Screen Shot 2019-09-05 at 14 08 41
Current: 

Screen Shot 2019-09-05 at 14 08 51

File: _layouts/post.html [hunk 22]
```diff
@@ -1,58 +1,66 @@
...
+          <button type="button" class="btn deploy" ga-event-category="guide-cta" ga-event-action="guide-deploy-cta">Deploy for me</button>
+          <button type="button" class="btn btn-link dismiss-cta" -event-category="guide-cta" ga-event-action="guide-cta-dismiss">Dismiss</button>
+        </div>
+      </div>
     </div>
   </div>
 </div>
+<div class="post-bg comments"></div>
+{% include newsletter.html %}
```
Comment (Etiene) [inline _layouts/post.html]: Maybe until the newsletter is perfected, since we already have a newsletter link at the footer, I would comment this out, create a JIRA card and work on the newsletter on a future PR

File: assets/js/guides.js (hunk 14 above)
Comment (Etiene) [reply assets/js/guides.js]: (this is another fix that could go in a new newsletter-only PR)

File: assets/js/guides.js (hunk 15 above)
Comment (Etiene) [reply assets/js/guides.js]: (another one that could go in a new newsletter-only PR)

File: assets/js/guides.js (hunk 16 above)
Comment (Etiene) [reply assets/js/guides.js]: (same as above)

File: assets/js/search.js [hunk 23]
```diff
@@ -0,0 +1,218 @@
...
+ * Javascript specially for the IaC Library page.
+ */
+
+(function () {
+
+  // Returns a function, that, as long as it continues to be invoked, will not
+  // be triggered. The function will be called after it stops being called for
+  // N milliseconds. If `immediate` is passed, trigger the function on the
+  // leading edge, instead of the trailing.
+  // From: https://davidwalsh.name/javascript-debounce-function
```
Comment (Etiene) [inline assets/js/search.js]: I didn't understand anything about this comment until I went to the website, which contains a very good examplation: "ensures a given task doesn't fire so often that it bricks browser performance". I'd add this to the comment.
```suggestion
  // Ensures a given task doesn't fire so often that it bricks browser performance.
  // From: https://davidwalsh.name/javascript-debounce-function
```

File: assets/js/search.js (hunk 18 above)
Comment (Etiene) [reply assets/js/search.js]: This one looks like it's a TO-DO mentioned in the PR description

File: assets/js/search.js (hunk 19 above)
Comment (Etiene) [reply assets/js/search.js]: > Not sure how this works at all? T
it "works" because it's an else if, so it already checked the first option and this is the only other option and an assignment always returns true 😅 So this reads as `if (searchEntry.type == 'libraryEntries') {} else {}`. Still bug in the code, but this is why it coincidentally works

File: assets/js/search.js [hunk 24]
```diff
@@ -0,0 +1,218 @@
...
+      clearTimeout(timeout);
+      timeout = setTimeout(later, wait);
+      if (callNow) func.apply(context, args);
+    };
+  }
+
+  function showItemsCount(totalCount, numSubmodules) {
+    if (searchEntry.type == 'libraryEntries') {
+      if (totalCount > 0 && numSubmodules > 0) {
+        $('#search-results-count').html("<strong>" + totalCount + "</strong> repos (~<strong>" + numSubmodules + "</strong> modules)");
```
Comment (Etiene) [inline assets/js/search.js]: Can we add `<strong>` through CSS and use the least tags as possible here? Can we also just show `X found` for both library page and guides page so we can remove these ifs here?

File: _includes/search.html [hunk 25]
```diff
@@ -0,0 +1,36 @@
...
+                <div class="tags"></div>
+            </div>
+        </div>
+    </div>
+    <div class="text-center">
+            <div class="cloud-filter">
+                <button id="aws" value="aws" ga-event-category="guide-cloud-filter" ga-event-action="filter-aws"><img  src="/assets/img/logos-technologies/aws-logo.png" alt="aws logo"></button>
+                <button id="gcp" value="gcp" ga-event-category="guide-cloud-filter" ga-event-action="filter-gcp"><img src="/assets/img/logos-technologies/gcp-logo.png" alt="aws logo"></button>
+                <button id="azure" value="azure" ga-event-category="guide-cloud-filter" ga-event-action="filter-azure"><img src="/assets/img/logos-technologies/azure-logo.png" alt="aws logo"></button>
+            </div>
```
Comment (Etiene) [inline _includes/search.html]: The image quality of these logos doesnt look really good
Screen Shot 2019-09-05 at 14 07 56

File: _includes/footer.html (hunk 5 above)
Comment (Etiene) [inline _includes/footer.html]: The spacing and color of these elements look off. The label needs a lot more margin

Expected:
Screen Shot 2019-09-05 at 14 11 40

Current:
Screen Shot 2019-09-05 at 15 40 34

File: _includes/search.html (hunk 25 above)
Comment (Etiene) [reply _includes/search.html]: Also it still needs to be marked as active after it's clicked (and not just on hover)
Screen Shot 2019-09-05 at 14 13 02

File: assets/js/search.js [hunk 26]
```diff
@@ -0,0 +1,218 @@
...
+
+  function displayFilterTags() {
+    if (searchEntry.type == 'guideEntries') {
+
+      //Filters the tags from the array of objects and flattens it out since it returns an array of arrays
+      let tags = searchEntry.entries.map(entry => entry.tags.split(',')).reduce((a, b) => a.concat(b), []);
+
+      tags.filter((tag, index) => {
+        if ((tags.indexOf(tag) === index) && (tag.length != "")) {
+          $('.tags').append(`<label class="checkbox"><input value=${tag} type="checkbox">${tag}</label>`);
```
Comment (Etiene) [inline assets/js/search.js]: Filter tags look misaligned and are not properly capitalized 
Screen Shot 2019-09-05 at 14 14 26

File: _includes/newsletter.html [hunk 27]
```diff
@@ -0,0 +1,23 @@
+<div class="guides-section-white">
+    <div class="row text-center">
+        <div class="col-md-6 col-md-offset-3 newsletter" >
+            <img src="/assets/img/guides/img-newsletter.png" alt="newsletter_image" class="img-center"><br>
+            <label>Newsletter</label>
+            <p>Subscribe to the Gruntwork Newsletter to get notified when updates and new guides are released. No spam. Ever.</p>
+            <form id="newsletter-form" class="col-xs-10 col-xs-offset-1" method="post" action="//gruntwork.us13.list-manage.com/subscribe/post?u=8ccaad0747ac13944cf318e0f&amp;id=786bd365d1">
+                <div class="input-group">
+                    <input type="text" id="email" class="form-control guides-input-box" placeholder="Your Email Address">
```
Comment (Etiene) [inline _includes/newsletter.html]: Some things off with the newslatter info, but mainly the vertical alignment of the email address. As I mentioned in other comments though, I would remove the newsletter from this article page and fix these issues on a future PR.


Current: 
Screen Shot 2019-09-05 at 14 11 18

Expected:
Screen Shot 2019-09-05 at 15 51 17

**Comment by Etiene** [comment]
There seems to be something off with the nav bar and the width of the full page. The nav bar doesnt look exactly the same size as when we switch bad to `How it works` for example. And the page always has a little bit more of width than the full screen and always has an horizontal scroll bar which makes the page flicker weirdly when scrolling (this might be the root cause of the navbar looking different, so I would look into this problem first). It happens on both the guide list and the guide post page.

Screen Shot 2019-09-05 at 14 07 43

File: _layouts/post.html [hunk 28]
```diff
@@ -1,58 +1,66 @@
...
-  <div class="container-fluid section-ghostwhite">
-    <div class="col-md-3">
-      <div id="all-guides">
-        <label class="pull-right" >
-          <a href="/guides">&lt; All Guides</a>
-        </label>
+<div  class="container-fluid post-bg">
+  <div class="post-detail">
+    <div class="col-sm-2-5">
+      <div id="toc">
```
Comment (Etiene) [inline _layouts/post.html]: There are some weird behaviours with this toc. 

1. Some items are not appearing aligned in the list
2. The marker doesnt correpond to the one on design when highlighted (should be a bullet point instead of an arrow)
3. The behavior of expanding on hover is seems really undesirable. I personally found it annoying to use. Is this really the intention of the design? Could we ask Sam about this one, please? If you did already, what did she say?

Screen Shot 2019-09-05 at 14 10 35

File: assets/js/guides.js (hunk 16 above)
Comment (hngerebara) [reply assets/js/guides.js]: No it deosn't

File: _layouts/post.html (hunk 21 above)
Comment (hngerebara) [reply _layouts/post.html]: I didn't continue with this because there was still discussion around if it should be taken off or not. Jim confirmed @ No 8 that this should be removed

File: _includes/search.html (hunk 25 above)
Comment (hngerebara) [reply _includes/search.html]: It marks as on hover, focus and active.

File: _includes/newsletter.html (hunk 27 above)
Comment (hngerebara) [reply _includes/newsletter.html]: This is a screen resolution issue. It displays properly on mine and other screens below mine. Will fix in my next PR for screen resolutions above mine

File: _layouts/post.html (hunk 28 above)
Comment (hngerebara) [reply _layouts/post.html]: I never asked her about this but since on hover seems annoying, I think it would be better implementing it on click instead

**Comment by Etiene** [comment]
@hngerebara Sorry, just noticed I repeated some of Jim's comments since some of them don't appear when browsing through "files changed" and I didn't see it. Please ignore the duplicates.

@brikis98 
> 3. The outline on the left side should highlight and show a dot next to the item you're looking at. This should update and auto expand as you scroll down.
>4. The carrots on the
>5. Title in breadcrumb doesn't match title of post.

The carrots on the...?

File: _layouts/post.html (hunk 21 above)
Comment (Etiene) [reply _layouts/post.html]: Ok!

File: _includes/search.html (hunk 25 above)
Comment (Etiene) [reply _includes/search.html]: I dont think the active one is working, it disappears right after you click it

File: _includes/footer.html (hunk 5 above)
Comment (hngerebara) [reply _includes/footer.html]: I had previously styled button links which I wanted to reuse. I can change it

File: _posts/2019-08-26-how-to-use-gruntwork-service-catalog.adoc (hunk 12 above)
Comment (hngerebara) [reply _posts/2019-08-26-how-to-use-gruntwork-service-catalog.adoc]: Yes we can define this in the config

File: assets/css/global.less (hunk 13 above)
Comment (hngerebara) [reply assets/css/global.less]: Styling for the placeholders within the input box to cater for various browsers https://github.com/gruntwork-io/gruntwork-io.github.io/pull/147/files/6da8b60d99762bbf2bc0fd6c78719c20420c8dbf#diff-1e69babe6c50a615d3a6ef7b48cf7484R888

**Comment by brikis98** [comment]
> The carrots on the...?

lol, got distracted while typing... Removed/fixed.

**Comment by Etiene** [comment]
Ok, let's trim the pending comments to wrap this app


From Jim:
Issues found while browsing the site via ngrok:

/guides page:

~~The page content stretches too wide on a high resolution (1920x1080), especially the search box and list of guides themselves. We probably need to set a max-width.~~

IN PROGRESS: There's no cloud filter selected by default... And once you select one, there's no way to deselect it. We probably want to have AWS selected by default (which I think is in the InVision design).

The categories should not disappear when you search or select a filter. The categories should always be there, but perhaps become grayed out when there are no matching results in those categories.

The category you're looking at should be highlighted ~~and have a dot next to it.~~ It should auto update as you scroll down.

~~Thumbnail image size should have max-width and max-height so it doesn't end up being too big. E.g., The "How to configure a production-grade AWS account structure" thumbnail ends up too big, so there isn't enough padding around it.~~

Filters seem to be modified to title-case (e.g., Aws vs AWS). That doesn't seem right?

Checkbox style in filter panel doesn't match InVision.

/guides/<category>/<guide> page:

Subheadings should have anchor links, as the rest of the site, and as on GitHub.

~~Line height in outline on left side is wrong, so long headings wrap and have too much vertical space.~~

~~The outline on the left side should highlight~~ and show a dot next to the item you're looking at. This should update and auto expand as you scroll down.

Title in breadcrumb doesn't match title of post.

The underline shouldn't include the slash in the underlined link in the breadcrumb

Change the line-height on the post title; right now, when it wraps, there is too much space betwene the lines.

~~Let's hide the "deploy for me" CTA for now. We need more thinking on this and can come back to it in another PR.~~

~~Let's hide the "xxx min read" for now. The numbers are huge and may scare people off.~~

Figure labels are not rendered correctly. They should use a smaller font and be close to the image... But have margin below them to move down the text around them. Same thing for the labels for code blocks.

The call outs don't look quite right. Instead of "Important" text, it should be an exclamation mark. The callout needs more vertical margin around it to separate it from the surrounding text. Not in the design: callout text should be left aligned; if the callout has a title, it should be left-aligned and bolded, with a little margin below it.

IN PROGRESS: Table styling looks wrong. It's faded out for some reason? Column headings have weird underlines?

~~Code blocks should not be indented on the left and right.~~

~~There seems to be random whitespace in some of the posts. E.g,. The "How to use the Gruntwork Service Catalog" post has it all over. What's causing that?~~

TO DISCUSS WITH SAM AS MOCKS SHOW INDENTATION: Lists (whether numbered, unnumbered, or definition) should probably not be indented, as we use a lot of lists, and it makes the text hard to follow as you scroll as it keeps jumping left and right. 

Make tags clickable

posts

Do we need to specify a permalink for every post? Seems error prone. Could we define this in the layout or Jekyll config instead?



From me:
Grunty space on "Ready to hand off work"

Image quality of cloud filters

~~page width~~

~~nav bar font~~

newsletter comments

**Comment by Etiene** [comment]
Prioritizing what's left (results of Etiene & Hope zoom)

# Improvements and Bug fixes necessary for launching this
(ordered list)
1. IN PROGRESS: [guides index] There's no cloud filter selected by default... And once you select one, there's no way to deselect it. We probably want to have AWS selected by default (which I think is in the InVision design).
2. IN PROGRESS: [guide post]  Table styling looks wrong. It's faded out for some reason? Column headings have weird underlines?
3. [guide post] Internal links on posts are not being rendered properly
4. [guide post] Figure labels are not rendered correctly. They should use a smaller font and be close to the image... But have margin below them to move down the text around them. Same thing for the labels for code blocks.
5. [guide post] Title in breadcrumb doesn't match title of post.
6. [guide post] Fix padding around "this module is for subscribers only" warning
7. [guides index] The categories should not disappear when you search or select a filter. The categories should always be there, but perhaps become grayed out when there are no matching results in those categories.
8. [guide post] The call outs don't look quite right. Instead of "Important" text, it should be an exclamation mark. The callout needs more vertical margin around it to separate it from the surrounding text. Not in the design: callout text should be left aligned; if the callout has a title, it should be left-aligned and bolded, with a little margin below it.


# Improvements and Bug fixes that can be reprioritized and worked on in parallel with adding more guides
(not fully ordered list yet)
[guide post] Change the line-height on the post title; right now, when it wraps, there is too much space between the lines.
[guides index] The category you're looking should auto update as you scroll down.
[guides index] Filters seem to be modified to title-case (e.g., Aws vs AWS). That doesn't seem right?
[guides index] Checkbox style in filter panel doesn't match InVision.
[guide post] Subheadings should have anchor links, as the rest of the site, and as on GitHub.
[guide post] The underline shouldn't include the slash in the underlined link in the breadcrumb
[guide post] TO DISCUSS WITH SAM AS MOCKS SHOW INDENTATION: Lists (whether numbered, unnumbered, or definition) should probably not be indented, as we use a lot of lists, and it makes the text hard to follow as you scroll as it keeps jumping left and right. 
[guide post] Make tags clickable
[post doc file] Do we need to specify a permalink for every post? Seems error prone. Could we define this in the layout or Jekyll config instead?
[guide post] Grunty space on "Ready to hand off work"
[guide post] Finish newsletter and correct remaining comments
[general] put guides css outside of global.less
[guides index] categories on the side of guides index are a bit jumpy when scrolling and scrolling past footer section in a weird way
